drivers/serial/uart_nrfx_uarte.c

```diff
@@ -705,6 +705,12 @@ static void endrx_isr(struct device *dev)
 	NRF_UARTE_Type *uarte = get_uarte_instance(dev);
 
 	if (!data->async->rx_enabled) {
+		if (data->async->rx_buf == NULL) {
```

Comment:
I think this deserves a comment that such condition (ENDRX interrupt while `rx_buf` is `NULL` and `rx_enabled` is `false` can occur only after triggering the FLUSHRX task). Also a comment before triggering the FLUSHRX task that the final part of handling the RXTO event is placed here (in the ENDRX interrupt handler) would be a nice thing.

Reply:
Added requested comments